src/libraries/System.Security.Cryptography.Cose/tests/CoseTestHelpers.cs

```diff
@@ -281,11 +351,21 @@ internal static void AssertMultiSignMessageCore(
             AssertExtensions.SequenceEqual(msg.Encode(), encodedMsg);
         }
 
-        internal static int GetSignatureSize(AsymmetricAlgorithm key)
+        internal static int GetSignatureSize(IDisposable key)
         {
-            int size = (key.KeySize + 7) / 8;
+#pragma warning disable SYSLIB5006
+            if (key is MLDsa mldsa)
+            {
+                return mldsa.Algorithm.SignatureSizeInBytes;
+            }
+#pragma warning restore SYSLIB5006
+
+            AsymmetricAlgorithm? asymmetricKey = key as AsymmetricAlgorithm;
+            Assert.NotNull(asymmetricKey);
```

Comment:
May as well just hard cast.  If it fails, it'll tell you what the unexpected type is, which is better than "I expected not null, but got null".

```suggestion
            AsymmetricAlgorithm asymmetricKey = (AsymmetricAlgorithm)key;
```